### PR TITLE
Cart-Management Functionality

### PR DESCRIPTION
**Pull request by ark2002**
- Added Cart-Management Functionality.
- Added Cart-Services.
- Added Cart-Context.
- Added Badge on navbar to show number of items in cart.

Preview:

https://deploy-preview-7--astromarts.netlify.app/

recording

### CODE REVIEW

**Review by tushargoyal812 — COMMENTED** [review]
didn't find any mistake in your code.
liked your variable naming.
folder structure is good.

File: src/frontend/components/CartSummary/CartSummary.css [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    justify-content: space-between;
+    padding-bottom: 1rem;
+}
+
+.price__summary:nth-child(4) {
+    border-bottom: 1px solid #999999;
+}
+
+.price__summary:nth-child(5) {
+    border-bottom: 1px solid #999999;
```
Comment (chanchal-panpaliya) [inline src/frontend/components/CartSummary/CartSummary.css:30]: you can use - :root { --border-bottom:"1px solid #999999" } then border-bottom : var(--border-bottom) . it's save your time.

File: src/frontend/components/CartPageCard/CartPageCard.css [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+.cart__product--quantity-btn:hover {
+    background-color: var(--secondary-color);
+}
+
+.cart__product--quantity {
+    font-size: 1.6rem;
+    background-color: var(--hover-color);
+    width: 3rem;
+    text-align: center;
+    padding: 5px;
```
Comment (chanchal-panpaliya) [inline src/frontend/components/CartPageCard/CartPageCard.css:67]: use only rem or px .

**Review by chanchal-panpaliya — COMMENTED** [review]
astromart . count?? and I really love your project theme, "Astromart".

File: src/frontend/components/CartPageCard/CartPageCard.css (hunk 2 above)
Comment (ark2002) [reply src/frontend/components/CartPageCard/CartPageCard.css:67]: it was recommended by vikram to use px under 10 px

**Pull request merged**